openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -155,8 +177,8 @@ private void handleCodeGenerationMode(ToolContext toolContext,
             generateClient(toolContext, codeGeneratorConfig);
         } else {
             Constants.DiagnosticMessages error = Constants.DiagnosticMessages.WARNING_FOR_OTHER_GENERATION;
-            reportDiagnostics(error, String.format(error.getDescription(), mode),
-                    location);
+            toolContext.reportDiagnostic(createDiagnostics(error, String.format(error.getDescription(), mode),
```

Comment:
It seems like `createDiagnostics` method is called everytime we call `toolContext.reportDiagnostic`. Can't we just get the required parameters as the input for `toolContext.reportDiagnostic` and call the `createDiagnostics` method from within that ?

Reply:
Address with https://github.com/ballerina-platform/openapi-tools/pull/1591/commits/55952e8d22e6b21c0dcdc4e50aaf6e540f2fa778